src/apps/opening-hours/OpeningHoursHelpers.ts

```diff
@@ -0,0 +1,80 @@
+import dayjs from "dayjs";
+import "dayjs/locale/da";
+import weekOfYear from "dayjs/plugin/weekOfYear";
+import { DplOpeningHoursListGET200Item as OpeningHoursItem } from "../../core/dpl-cms/model";
+
+dayjs.extend(weekOfYear);
+export const getNextWeek = (date: Date): Date => {
+  return dayjs(date).add(1, "week").toDate();
+};
+
+export const getPreviousWeek = (date: Date): Date => {
+  return dayjs(date).subtract(1, "week").toDate();
+};
+
+export const getWeekDisplay = (date: Date): string => {
+  const week = dayjs(date).week();
+  const year = dayjs(date).year();
+
+  return `Uge ${week}, ${year}`;
+};
+
+const capitalizeString = (string: string): string => {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+};
+
+export const formatDateToWeekday = (date: Date): string => {
+  const formattedAsWeekday = dayjs(date).locale("da").format("dddd");
+  const capitalizedWeekday = capitalizeString(formattedAsWeekday);
+
+  return capitalizedWeekday;
+};
+
+export const formatDate = (date: Date): string => {
+  return dayjs(date).locale("da").format("YYYY-MM-DD");
+};
+
+export const getWeekStartAndEndDate = (date = new Date()) => {
+  const start = dayjs(date).locale("da").startOf("week").toDate();
+  const end = dayjs(date).locale("da").endOf("week").toDate();
+
+  return { start, end };
+};
+
+export type GroupedOpeningHours = Array<{
+  day: string;
+  date: string;
+  openingHourEntries: OpeningHoursItem[];
+}>;
+
+export const groupOpeningHoursByWeekday = (
+  startDate: Date,
+  endDate: Date,
+  openingHours: OpeningHoursItem[]
+): GroupedOpeningHours => {
+  const startDay = dayjs(startDate);
+  const endDay = dayjs(endDate);
+  let currentDay = startDay;
+
+  // Generate array of days between start and end dates
+  const allDays = [];
+  while (currentDay.isBefore(endDay) || currentDay.isSame(endDay, "day")) {
```

Comment:
```suggestion
  while (currentDay.isBefore(endDay.add(1, "day")) {
```

You could also look into [Durations in dayjs](https://day.js.org/docs/en/durations/durations).

Reply:
This almost worked. 

I ended up refactoring to this 

```
 while (!currentDay.isAfter(endDay)) {
    allDays.push(currentDay.toDate());
    currentDay = currentDay.add(1, "day");
  }
  
 ```